### DISCUSSION TITLE
Environment-specific properties question

### DISCUSSION THREAD

**Discussion by jkchua** [getgauge/gauge#2260] [General]
Hello!

I am currently exploring the part of Gauge where I would like to be able to setup environment-specific variables and use them in the code to reduce hard-coded pieces. Say for example in the small user.properties snippet below:

```
# Sets the URL of the SIT environment of the API.
base_api_url = "<api endpoint>"

# Flag that denotes that the environment is lower than production.
is_lower_environment = true
```

The problem is, I do not know if by default, aside from the default.properties and java.properties, the selected environment (e.g. SIT) automatically loads the user.properties (_the snippet above_). 

- If they are automatically loaded without writing code, how do I access the environment-specific variables? 
- If they are not automatically loaded, how do I identify in the code what environment I am currently executing to pull out the properties?

I'm using Java with Maven by the way - if it helps.

**Comment by sriv** [comment]
see https://docs.gauge.org/configuration.html for details around configuration and environment variables.

In your specific example:

- if you place any `.properties` file under the `env/default` it gets loaded as an environment variable for gauge, and is passed on to the runner which executes your test code. You should be able to access it using `System.getenv` method.
- you can have multiple environments in gauge, i.e. you can have `env/ci`, `env/qa` etc and specify which environment to use by setting the `--env` flag.
- `default` env is available to all env, so you can put common properties in `env/default` and environment specific property in the relevant subdirectories.

Hope this helps.

**Reply by jkchua** [reply]
Yay! Thank you again @sriv for your help. It worked on my end and was able to make some parts of the codebase dynamic. System.getenv was the answer and the thoughts on other points helped clarify the documentation further. 😊
